pkgs/development/python-modules/pytest-faulthandler/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools_scm
+, pytest
+, pytest-mock
+, pythonOlder
+, faulthandler
+}:
+
+buildPythonPackage rec {
+  pname = "pytest-faulthandler";
+  version = "1.5.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "bf8634c3fd6309ef786ec03b913a5366163fdb094ebcfdebc35626400d790e0d";
+  };
+
+  buildInputs = [ setuptools_scm ];
+  checkInputs = [ pytest-mock ];
+  propagatedBuildInputs = [ pytest ] ++ lib.optional (pythonOlder "3.0") faulthandler;
```

Comment:
Don't propagate pytest, it's a special case. Plugins should only use it for build time of itself.